src/Microsoft.ApplicationInsights/Extensibility/Implementation/ApplicationId/DictionaryApplicationIdProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.ApplicationId
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Application Id Provider that holds a dictionary of Instrumentation Keys to Application Ids.
+    /// </summary>
+    public class DictionaryApplicationIdProvider : IApplicationIdProvider
+    {
+        /// <summary>
+        /// Gets or sets a dictionary of Instrumentation Keys to Application Ids
+        /// </summary>
+        public Dictionary<string, string> Defined { get; set; }
```

Comment:
Consider typing this as `IReadOnlyDictionary<string,string>` which allows for more flexibility.